.github/workflows/continuous-integration.yml

```diff
@@ -55,38 +34,8 @@ jobs:
           # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --statistics
 
-  analyse:
-    name: Analyse
-    needs: skip_duplicate
-    if: ${{ needs.skip_duplicate.outputs.should_skip == 'false' }}
-    runs-on: ubuntu-latest
-
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
-        with:
-          # We must fetch at least the immediate parents so that if this is
-          # a pull request then we can check out the head.
-          fetch-depth: 2
-
-      # If this run was triggered by a pull request event, then checkout
-      # the head of the pull request instead of the merge commit.
-      - run: git checkout HEAD^2
-        if: ${{ github.event_name == 'pull_request' }}
-
-      # Initializes the CodeQL tools for scanning.
-      - name: Initialize CodeQL
-        uses: github/codeql-action/init@v1
-        # Override language selection
-        with:
-          languages: python, javascript
-
-      - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v1
-
   test-pip:
     needs: lint
-    if: ${{ needs.skip_duplicate.outputs.should_skip == 'false' }}
     strategy:
       matrix:
         python-version: [ 3.7, 3.8, 3.9, "3.10", "3.11"]
```

Comment:
Add the packages you have version pinned in the `dependencies`. You can add future versions and skip them if they are not present yet.

Reply:
I've discovered that I could do something like this:
```
    strategy:
      matrix:
        python-version: [ 3.7, 3.8, 3.9, "3.10", "3.11"]
        markdown-it-py-version: [2]
        include:
          - python-version: 3.9
            markdown-it-py-version: null
          - python-version: 3.9
            markdown-it-py-version: 3
          - python-version: 3.9
            no-kernel: true
          - python-version: 3.9
            quarto: true
```
it's really fun!